src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -46,32 +47,38 @@ private IntakePosition(double value) {
 
     private final CANSparkMax intake;
     private final WPI_TalonSRX deploy;
-    private final DigitalInput limitSwitch;
 
-    private double intakePower = 0;
-    private boolean driverOverride = false;
+    private final DigitalInput topLimitSwitch;
+    private final Timer topTimer = new Timer();
+    private final DigitalInput bottomLimitSwitch;
+    private final Timer bottomTimer = new Timer();
+
+    private static final double TOP_LIMIT_DELAY = 0.01;
+    private static final double BOTTOM_LIMIT_DELAY = 0.01;
 
-    private static final double DELAY_LIMIT_RESET = 0.2;
-    private Double switchPressed = 0.0;
+    private double driverPower = 0;
+    private boolean driverOverride = false;
```

Comment:
what if we assumed that if driverpower isn't 0 then we want to override

Reply:
i really don't like having all these powers and overrides

Reply:
That works, but it means that we can't override with 0 power (something that RunIntakeCommand does for 2 seconds after driver input is 0). I guess we can remove `driverOverride` entirely since we're not using the intake camera to detect balls at all.